### PR TITLE
Add Databricks PySpark job example DAG task, update cluster config

### PR DESCRIPTION
**Pull request by dmoore247**
This pull request adds a Databricks PySpark job as an Airflow Task to the existing Databricks example. 
the PySpark job task was difficult to find for me so I'm adding the short example I developed back to the community (and my future self). This example also adds a parameter and an egg file library for richness and completeness.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/databricks/example_dags/example_databricks.py [hunk 1]
```diff
@@ -71,4 +72,17 @@
         libraries=[{'jar': 'dbfs:/lib/etl-0.1.jar'}],
     )
     # [END howto_operator_databricks_named]
-    notebook_task >> spark_jar_task
+
+    # [START howto_operator_databricks_pyspark]
```
Comment (josh-fell) [inline airflow/providers/databricks/example_dags/example_databricks.py:76]: These `# [START/END howto_operator_...]` tags are used as references in the operator documentation so the docs can display code snippets of the related example DAG between these `START` and `END` tags. Using these tags keeps the examples and the operator docs in sync. Would you be willing to add some context around this new task and how it relates to the `DatabricksSubmitRunOperator`?

You can check out how similar tags are used for the existing operators in the Databricks operator documentation [here](https://github.com/apache/airflow/edit/main/docs/apache-airflow-providers-databricks/operators.rst).

Comment (potiuk) [reply airflow/providers/databricks/example_dags/example_databricks.py:76]: Oh yeah That should be indeed updated in the Databricks howto: https://airflow.apache.org/docs/apache-airflow-providers-databricks/stable/operators.html

Comment (dmoore247) [reply airflow/providers/databricks/example_dags/example_databricks.py:76]: @josh-fell @potiuk Thanks for the pointer... I'll see about making that update.

Comment (dmoore247) [reply airflow/providers/databricks/example_dags/example_databricks.py:76]: My colleague has an open [PR](https://github.com/apache/airflow/pull/19156) that would overlap with this one. I'd like to withdraw this PR and re-submit once the other PR is merged.

**Pull request closed without merging**